control_plane/storcon_cli/src/main.rs

```diff
@@ -0,0 +1,643 @@
+        NodeAvailabilityWrapper, NodeDescribeResponse, ShardSchedulingPolicy,
+        TenantDescribeResponse, TenantPolicyRequest,
+    },
+    models::{
+        LocationConfig, ShardParameters, TenantConfig, TenantCreateRequest,
+        TenantShardSplitRequest, TenantShardSplitResponse,
+    },
+    shard::{ShardStripeSize, TenantShardId},
+};
+use pageserver_client::mgmt_api::{self, ResponseErrorMessageExt};
+use reqwest::Url;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use utils::id::{NodeId, TenantId};
+
+use pageserver_api::controller_api::{
+    NodeConfigureRequest, NodeRegisterRequest, NodeSchedulingPolicy, PlacementPolicy,
+    TenantLocateResponse, TenantShardMigrateRequest, TenantShardMigrateResponse,
+};
+
+#[derive(Subcommand, Debug)]
+enum Command {
+    NodeRegister {
+        #[arg(long)]
+        node_id: NodeId,
+
+        #[arg(long)]
+        listen_pg_addr: String,
+        #[arg(long)]
+        listen_pg_port: u16,
+
+        #[arg(long)]
+        listen_http_addr: String,
+        #[arg(long)]
+        listen_http_port: u16,
+    },
+    NodeConfigure {
+        #[arg(long)]
+        node_id: NodeId,
+
+        #[arg(long)]
+        availability: Option<NodeAvailabilityWrapper>,
+        #[arg(long)]
+        scheduling: Option<NodeSchedulingPolicy>,
+    },
+    TenantPolicy {
+        #[arg(long)]
+        tenant_id: TenantId,
+        #[arg(long)]
+        placement: Option<PlacementPolicy>,
+        #[arg(long)]
+        scheduling: Option<ShardSchedulingPolicy>,
+    },
+    Nodes {},
+    Tenants {},
+    TenantCreate {
+        #[arg(long)]
+        tenant_id: TenantId,
+    },
+    TenantDelete {
+        #[arg(long)]
+        tenant_id: TenantId,
+    },
+    TenantShardSplit {
+        #[arg(long)]
+        tenant_id: TenantId,
+        #[arg(long)]
+        shard_count: u8,
+        #[arg(long)]
+        stripe_size: Option<u32>,
+    },
+    TenantShardMigrate {
+        #[arg(long)]
+        tenant_shard_id: TenantShardId,
+        #[arg(long)]
+        node: NodeId,
+    },
+    TenantConfig {
+        #[arg(long)]
+        tenant_id: TenantId,
+        #[arg(long)]
+        config: String,
+    },
+    ServiceRegister {
+        #[arg(long)]
+        http_host: String,
+        #[arg(long)]
+        http_port: u16,
+        #[arg(long)]
+        region_id: String,
+        #[arg(long)]
+        availability_zone_id: String,
+        #[arg(long)]
+        version: u64,
+        // {
+        //   "version": 1,
+        //   "host": "${HOST}",
+        //   "port": 6400,
+        //   "region_id": "{{ console_region_id }}",
+        //   "instance_id": "${INSTANCE_ID}",
+        //   "http_host": "${HOST}",
+        //   "http_port": 9898,
+        //   "active": false,
+        //   "availability_zone_id": "${AZ_ID}",
+        //   "disk_size": ${DISK_SIZE},
+        //   "instance_type": "${INSTANCE_TYPE}",
+        //   "register_reason" : "New pageserver"
+        // }
```

Comment:
What is this?

Reply:
Reference of what the rest of the struct looks like (gone now since I removed service-register)